### DISCUSSION TITLE
Plugin to append a hash to upload filenames

### DISCUSSION THREAD

**Discussion by jeanbmar** [payloadcms/payload#301] [Show and tell]
Hey there,

I've released [payload-hash-upload](https://github.com/jeanbmar/payload-hash-upload), a plugin to append a hash to upload filenames. 
We use the plugin to ensure unique names without harming SEO, and to optimize our CDN image caching strategy.

**Comment by DanRibbens** [comment]
Great work! You are solving a real problem that many users may deal with.
I am wondering if this functionality should be implemented directly in Payload.

I'm Looking over the repo now; you've done an amazing job!

**Reply by jeanbmar** [reply]
Thank you! Feel free to reuse any part of the code in Payload.

**Reply by DanRibbens** [reply]
Alright, we'll add this to our backlog as an item to revisit when we have bandwidth and prioritize based on feedback.

Thanks!
